Source/Scripts/SourceToBinary.cs

```diff
@@ -67,12 +70,31 @@ public static async Task Handle(string inputFile, TextWriter outputFile) {
     await outputFile.WriteAsync(output);
     await outputFile.FlushAsync();
   }
+
+  private static void DeleteSources(ProgramParseResult parseResult) {
+    var toVisit = new Stack<INode>();
+    toVisit.Push(parseResult.Program);
+    while (toVisit.Any()) {
+      var current = toVisit.Pop();
+      foreach (var child in current.Children) {
+        if (child == null) {
+          throw new Exception();
+        }
+        toVisit.Push(child);
+      }
+
+      if (current is NodeWithOrigin withOrigin && withOrigin.Origin != Token.NoToken) {
+        var token = new Token(-1, -1) { Uri = withOrigin.Origin.Uri };
```

Comment:
Good bang for the buck coverage with this approach. For the record I'm still a bit worried the hanging I observed when locations were different but overlapped isn't covered.